deepchem/splits/splitters.py

```diff
@@ -1408,28 +1456,25 @@ def split(
       Each indices is a list of integers.
     """
     np.testing.assert_almost_equal(frac_train + frac_valid + frac_test, 1.)
-    scaffold_sets = self.generate_scaffolds(dataset)
```

Comment:
Yes, it would. It's random already, but I'm adding a possibility to manage that randomness.
Because of it, each test that uses `ScaffoldSplitter` will have slightly different splits, so the result metrics could be different. That is why some tests are failed.

Reply:
I believe `ScaffoldSplitter` isn't random at present. The sort is deterministic (https://github.com/deepchem/deepchem/blob/master/deepchem/splits/splitters.py#L1461) and based on size of scaffold splits. 

A change to the behavior of scaffold splitter that introduces new randomness would likely break existing pipelines that expect scaffold split to maintain stability across runs. It might be better as a change to explicitly document and test that `ScaffoldSplitter` is deterministic.